activerecord/lib/active_record/relation.rb

```diff
@@ -953,6 +953,10 @@ def _increment_attribute(attribute, value = 1)
       end
 
       def exec_queries(&block)
+        if respond_to?(:proxy_association) && proxy_association.violates_strict_loading?
+          Base.strict_loading_violation!(owner: proxy_association.owner.class, reflection: proxy_association.reflection)
```

Comment:
Curious if we might also need to check the proxy_association for strict_loading_value on line 973? In case a strict_loading setting needs to be carried down to a specific record in the result. Not very familiar with this part of the code

Reply:
Hmm good catch, I think you're right, I'll see if I can write a failing test.

Reply:
On second thought, I don't think this is an issue. If we're at this stage we know the query method is being called on an association and will be executed. So if strict loading is violated we'll always raise / log, else the returned records are outside of the strict loading "scope". At least that's what I can tell by the lack of tests for nested records beyond the first association (and second in the case of `has_many` + `:through`).